src/day12/Main.kt

```diff
@@ -0,0 +1,98 @@
+package com.zpthacker.aoc21.day12
+
+import com.zpthacker.aoc21.getInput
+
+private var connections = mutableMapOf<Cave, Set<Cave>>()
+private var caves = mutableMapOf<String, Cave>()
+
+fun main() {
+    val input = getInput(12).trim()
+    val lines = input.split("\n")
+    lines.fold(caves) { acc, caveLine ->
```

Comment:
I just realized I don't have to use `fold` here, this was vestigial of my initial bad solution. Same for the other place I use it.